libkbfs/md_journal_test.go

```diff
@@ -897,6 +899,53 @@ func testMDJournalClear(t *testing.T, ver MetadataVer) {
 	flushAllMDs(t, ctx, signer, j)
 }
 
+func testMDJournalClearPendingWithMaster(t *testing.T, ver MetadataVer) {
+	_, _, id, signer, ekg, bsplit, tempdir, j := setupMDJournalTest(t, ver)
+	defer teardownMDJournalTest(t, tempdir)
+
+	firstRevision := MetadataRevision(10)
+	firstPrevRoot := fakeMdID(1)
+	mdCount := 10
+
+	_, prevRoot := putMDRangeHelper(t, ver, id, signer, firstRevision,
+		firstPrevRoot, mdCount, j.uid,
+		func(ctx context.Context, md *RootMetadata) (MdID, error) {
+			return j.put(ctx, signer, ekg, bsplit, md, true)
+		})
+
+	putMDRange(t, ver, id, signer, ekg, bsplit,
+		firstRevision+MetadataRevision(mdCount), prevRoot, mdCount, j)
+
+	ctx := context.Background()
+
+	err := j.convertToBranch(
+		ctx, PendingLocalSquashBranchID, signer, kbfscodec.NewMsgpack(), id,
+		NewMDCacheStandard(10))
+	require.NoError(t, err)
+	require.NotEqual(t, NullBranchID, j.branchID)
+
+	bid := j.branchID
+
+	// Clearing the correct branch ID should clear just the last
+	// half of the journal and reset the branch ID.
+	err = j.clear(ctx, bid)
```

Comment:
For what it's worth, in practice I'm not sure this would ever happen, because after this change, `mdJournal.clear()` is only ever called via `MDOps.PruneBranch()`, which isn't really used anymore except in unstaging.  But worth checking anyway I guess, for that case.